src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -1139,5 +1146,162 @@ public Stream GetTempStream()
                 throw new NotSupportedException("GetTempStream is never used in CounterActions. Shouldn't happen");
             }
         }
+
+        private class SubscriptionActions : ISubscriptionActions
+        {
+            private readonly DocumentDatabase _database;
+            private MergedBatchPutSubscriptionCommand _command;
+            private MergedBatchPutSubscriptionCommand _prevCommand;
+            private Task _prevCommandTask = Task.CompletedTask;
+            private readonly Sparrow.Size _enqueueThreshold;
+            private readonly Logger _log;
+
+            public SubscriptionActions(DocumentDatabase database, Logger log)
+            {
+                _database = database;
+                _log = log;
+                _enqueueThreshold = new Sparrow.Size(
+                    (sizeof(int) == IntPtr.Size || database.Configuration.Storage.ForceUsing32BitsPager) ? 2 : 32,
+                    SizeUnit.Megabytes);
+                _command = new MergedBatchPutSubscriptionCommand(database, log);
+            }
+
+            public void Dispose()
+            {
+                FinishBatchOfSubscriptions();
+            }
+
+            public void WriteSubscription(SubscriptionState subscriptionState)
+            {
+                _command.Add(subscriptionState);
+                HandleBatchOfSubscriptionssIfNecessary();
+            }
+
+            private void HandleBatchOfSubscriptionssIfNecessary()
+            {
+                var prevDoneAndHasEnough = _command.Context.AllocatedMemory > Constants.Size.Megabyte && _prevCommandTask.IsCompleted;
+                var currentReachedLimit = _command.Context.AllocatedMemory > _enqueueThreshold.GetValue(SizeUnit.Bytes);
+
+                if (currentReachedLimit == false && prevDoneAndHasEnough == false)
+                    return;
+
+                var prevCommand = _prevCommand;
+                var prevCommandTask = _prevCommandTask;
+
+                var commandTask = _database.TxMerger.Enqueue(_command);
```

Comment:
This is wrong. We should be invoking the `PutSubscription` here on each one of them, not trying to ship it to the Tx Merger and invoking it from there.
That is a bad idea.